### ISSUE TITLE
[Bug]: Wrong result got from OpenVINO for ReduceSum with Add compared with PyTorch

### ISSUE THREAD

**Issue by jiaqiuji** [openvinotoolkit/openvino#23616]
### OpenVINO Version

2024.0.0-14509-34caeefd078-releases/2024/0

### Operating System

Ubuntu 20.04 (LTS)

### Device used for inference

CPU

### Framework

PyTorch

### Model used

N/A

### Issue description

Got wrong accuracy from OpenVINO compared to torch, using the following short reproducer model. The problem might be related to an optimization inside CPU plugin, since using GPU/NPU it returned correct results.

### Step-by-step reproduction

Using the following script to reproduce:
```python
import torch
import openvino as ov

class TestModel(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.weight = torch.rand((1536, 1, 4), dtype=torch.float32)
        self.bias = torch.rand((1536,), dtype=torch.float32)
    
    def forward(self, x: torch.Tensor):
        return torch.sum(x * self.weight.squeeze(1), dim=-1) + self.bias

# input
input_data = torch.randn(1, 1536, 4)
# model
torch_model = TestModel()
ov_model = ov.convert_model(torch_model, example_input=[input_data])
ov_compiled_model = ov.compile_model(ov_model, device_name="CPU")
# infer
print("torch result:", torch_model(input_data).numpy())
print("ov result:", ov_compiled_model([input_data])[0])
```

### Relevant log output

```shell
torch result: [[ 0.58234537 -0.29093838  0.14263596 ...  0.6258036   0.70492244  1.0191829 ]]
ov result: [[ 0.58234537 -0.2700194   0.4450267  ...  0.03057986  0.33239797  0.8386613 ]]
```

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by mvafin** [comment]
Since GPU provide good accuracy the model is most likely correct, so the issue must be in CPU plugin. Assigned CPU team

**Comment by meiyang-intel** [comment]
Can reproduce the issue.

**Comment by avitial** [comment]
Ref. 138063

**Comment by yuxu42** [comment]
#23970

**Comment by avitial** [comment]
Closing this, PR addressing the issue has been merged to master. Feel free to reopen to ask any questions related to this topic.
